iroha/src/lib.rs

```diff
@@ -293,9 +306,58 @@ impl Iroha {
             sumeragi_message_handle,
             tx_handle,
             block_sync_message_handle,
+            genesis_block_handle,
         );
         Ok(())
     }
+
+    /// To wait for active peers(this is provisional code)
+    /// Wait 500ms, 1000ms, 1500ms, ...
+    async fn wait_for_active_peers(
+        sorted_peers: &[PeerId],
+        myself_peer_id: &PeerId,
+    ) -> Result<(), String> {
+        log::info!("Waiting for active peers");
+        let retry_count = 10;
+        let retry_duration_ms = 500;
+        let mut failed_to_connect = true;
+        for peer in sorted_peers.iter() {
+            for i in 0..=retry_count {
+                if peer == myself_peer_id {
+                    break;
+                }
+                let target = httpUri::Builder::new()
+                    .scheme("http")
+                    .authority(peer.address.as_str())
+                    .path_and_query(uri::HEALTH_URI)
+                    .build()
+                    .expect(&format!("Invalid peer address {}", peer.address.as_str()));
+                match isahc::get_async(target.to_string()).await {
```

Comment:
Sorry for the lack of explanation, it happens error
```
error[E0277]: the trait bound `isahc::http::Uri: std::convert::From<http::Uri>` is not satisfied
   --> iroha/src/lib.rs:335:23
    |
335 |                 match isahc::get_async(target).await {
    |                       ^^^^^^^^^^^^^^^^ the trait `std::convert::From<http::Uri>` is not implemented for `isahc::http::Uri`
    |
    = note: required because of the requirements on the impl of `std::convert::Into<isahc::http::Uri>` for `http::Uri`
    = note: required because of the requirements on the impl of `std::convert::TryFrom<http::Uri>` for `isahc::http::Uri`
```

Reply:
Isahc reexports http crate, so maybe it will be better to `use isahc::http::uri as http_uri` instead of `use http::uri as http_uri`

Reply:
You probably have a conflicting versions of http that isahc uses and the one used here.